### ISSUE TITLE
Arbitrary Neural Computation Graphs & Generalization to GPU and CPU also

### ISSUE THREAD

**Issue by chaudharyachint08** [mathemakitten/gradient-checkpointing#1]
Hello there,

After looking into multiple repositories for Gradient checkpointing, I found out that you way appears to be best. I am an avid Keras enthusiast and would like to know if your method works for any keras layer and with any possible neural network architecture? and if so does it also work consistently across any hardware (CPU/GPU/TPU)?

**Comment by mathemakitten** [comment]
Hi! I've used this with Transformers, but I adapted this from a repo which was designed for image work and I expect it should work for most Keras layers. I've used it on CPU + TPU, not sure about GPU.
